src/test/java/com/example/petbuddybackend/controller/CareControllerIntegrationTest.java

```diff
@@ -493,5 +498,63 @@ void cancelCareByClient_WhenCareIsAlreadyAcceptedByCaretaker_ShouldThrowIllegalA
                 .andExpect(status().isBadRequest());
     }
 
+    @Test
+    @WithMockUser(username = "caretakerEmail")
+    void getCaretakerCares_ShouldReturnProperCares() throws Exception {
+        // Given
+        PersistenceUtils.addCare(careRepository, caretaker, client, animalRepository.findById("DOG").get());
+        // When and Then
+        mockMvc.perform(get("/api/care/caretaker-cares")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .header(ROLE_HEADER_NAME, Role.CARETAKER))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.content[0].clientStatus").value(CareStatus.ACCEPTED.name()))
+                .andExpect(jsonPath("$.content[0].caretakerStatus").value(CareStatus.PENDING.name()))
+                .andExpect(jsonPath("$.content[0].careStart").value(LocalDate.now().plusDays(2).toString()))
+                .andExpect(jsonPath("$.content[0].careEnd").value(LocalDate.now().plusDays(7).toString()))
+                .andExpect(jsonPath("$.content[0].description").value("Test care description"))
+                .andExpect(jsonPath("$.content[0].dailyPrice").value(50.00))
+                .andExpect(jsonPath("$.content[0].animalType").value("DOG"))
+                .andExpect(jsonPath("$.content[0].caretakerEmail").value("caretakerEmail"))
+                .andExpect(jsonPath("$.content[0].clientEmail").value("clientEmail"));
+    }
+
+    @Test
+    @WithMockUser(username = "caretakerEmail")
+    void getCaretakerCares_whenRoleIsClient_ShouldThrowUnauthorizedException() throws Exception {
+        // Given
+        PersistenceUtils.addCare(careRepository, caretaker, client, animalRepository.findById("DOG").get());
+        // When and Then
+        mockMvc.perform(get("/api/care/caretaker-cares")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .header(ROLE_HEADER_NAME, Role.CLIENT))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser(username = "caretakerEmail")
+    void getCaretakerCares_whenRoleIsNotProvided_ShouldThrowUnauthorizedException() throws Exception {
+        // Given
+        PersistenceUtils.addCare(careRepository, caretaker, client, animalRepository.findById("DOG").get());
+        // When and Then
+        mockMvc.perform(get("/api/care/caretaker-cares")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+    }
```

Comment:
ShouldThrowUnauthorizedException a sprawdzane jest isBadRequest

Reply:
a bo to z tym headerem, ze rzuca co innego nie Unauthorized